manim/mobject/geometry.py

```diff
@@ -422,6 +479,30 @@ def point_at_angle(self, angle):
 
 
 class Dot(Circle):
+    """A circle with a very small radius.
+
+    Parameters
+    ----------
+    point : Optional[:class:`float`]
+        The location of the dot.
+    radius : Optional[:class:`float`]
+        The radius of the dot.
```

Comment:
point and radius aren't optional if I am not mistaken.

Reply:
...this applies to quite a few other positions. If I unterstood it correctly a parameter is only otional, if you are alowed to set it to None. In case of a default value I don't think you should use Optional.

Reply:
They're optional because they have default values set in the constructor

Reply:
You should use `Optional[...]` if the parameter accepts `None` as an argument (i.e. the case when `None` is passed in is explicitly handled somewhere). Otherwise, use ``:class:`float`, optional`` which states that the argument isn't needed because there's a default -- but it can't be `None`.

In this case, you *CANNOT* pass in `None` for `point`, but you [*CAN*](https://github.com/ManimCommunity/manim/blob/c03c05f11497059f35c6f1966f5aef36a1dcb3b3/manim/mobject/geometry.py#L272-L273) for `radius`.